### PR TITLE
Implement page-per-dataclass with field linking.

### PR DESCRIPTION
**Pull request by jsirois**
The reflection infrastructure introduced to render the interpreter
provider page and parse the lift manifest is expanded to handle full
cross-referencing of `@dataclass` types as well as handling generating a
separate page per `@dataclass`. This is used to restructure
`providers.md` and fill out the skeleton of the full lift `manifest.md` 
doc tree.

Added as well are a few facilities to ensure documentation is present
and not broken:
+ Invalid cross-references produce CI errors during sphinx html gen.
+ Invalid external links produce CI errors during sphinx linkcheck.

### CODE REVIEW

**Comment by jsirois** [comment]
I'm going to merge this on green, publish the docsite and link here. Makes a bit more sense when you can see it and navigate.

**Comment by jsirois** [comment]
Alright, live here: https://science.scie.app/
You can see the link-checking in action here: https://github.com/a-scie/lift/actions/runs/5216144974/jobs/9414564648#step:5:124

File: docs/_ext/sphinx_science/toml.py [hunk 1]
```diff
@@ -0,0 +1,342 @@
...
+    @classmethod
+    def create(cls, value_type: TOMLType) -> Self:
+        return cls(label=f"Table[Key, {value_type.label}]", value_type=value_type)
+
+    value_type: TOMLType | None = None
+
+    def render_value(self, value: Any) -> str:
+        if self.value_type:
+            return repr({key: self.value_type.render_value(val) for key, val in value.items()})
+        return repr(value)
```
Comment (jsirois) [inline docs/_ext/sphinx_science/toml.py:69]: A personal requirement here was to divorce the docs from Python-ness. Science ships as a scie; so the user doesn't know Python, they know TOML, which is the Science lift manifest format. As such, I want the docs to just assume what the tool assumes - which is pretty much just TOML for the config and a subset of shells for when the documenting of the shell-completion comes around.

You can see all of this on the main lift doc here: https://science.scie.app/_/manifest/lift.html from default values to None / empty -> (Optional), etc. There should be no signs of Python-speak.

**Review by benjyw — COMMENTED** [review]
Neat! And eliding away the python-ness is a great touch.

**Pull request merged**